internal/bloblang/query/type_helpers.go

```diff
@@ -230,6 +231,8 @@ func IGetTimestamp(v any) (time.Time, error) {
 		return time.Parse(time.RFC3339Nano, string(t))
 	case string:
 		return time.Parse(time.RFC3339Nano, t)
+	case primitive.ObjectID:
```

Comment:
This is specifically needed for the use case? We shouldn't really be importing third party client libraries into `./internal/bloblang` as it'll then bring them into every single possible custom build.